cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -70,7 +74,7 @@ struct ResolvedChunk {
 
   bool IsNull() const { return array->IsNull(index); }
 
-  ViewType GetView() const { return array->GetView(index); }
+  LogicalValueType Value() const { return V::LogicalValue(array->GetView(index)); }
```

Comment:
We need to refactor/rename `GetViewType`, this was very confusing to read (not in any way that's your fault)